File: 백준/Silver/11053. 가장 긴 증가하는 부분 수열/가장 긴 증가하는 부분 수열.py
import sys

input = sys.stdin.readline

n = int(input())
nums = list(map(int, input().split()))

d = [1] * n

for i in range(1, n):
    for j in range(i):
        if nums[i] > nums[j]:
            d[i] = max(d[i], d[j] + 1)
        
            
            
print(max(d))

File: 백준/Silver/11053. 가장 긴 증가하는 부분 수열/가장 긴 증가하는 부분 수열.java
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.util.Arrays;

public class Main {
    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(br.readLine());

        int[] arr = Arrays.stream(br.readLine().split(" "))
                .mapToInt(Integer::parseInt)
                .toArray();

        int[] dp = new int[n];
        for(int i = 0; i < n; i++) {
            dp[i] = 1;
        }

        for(int i = 1; i < n; i++) {
              for(int j = 0; j < i; j++) {
                  if (arr[i] > arr[j]) {
                      dp[i] = Math.max(dp[i], dp[j] + 1);
                  }
              }
        }

        int answer = 0;
        for (int i : dp) {
            answer = Math.max(i, answer);
        }

        System.out.println(answer);
    }
}